glass/grf/_transformations.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+# type imports for NDArray[Any] cannot be moved into TYPE_CHECKING here
+# otherwise, the dispatch mechanism cannot resolve the class dynamically
+from typing import Any
+
+from numpy.typing import NDArray  # noqa: TC002
+
+from glass.grf._core import corr, dcorr, icorr
+
+
+@dataclass
+class Normal:
+    """
+    Transformation for normal fields.
+
+    .. math::
+
+       t(X) = X
+
+    This is the identity transformation.
+
+    """
+
+    def __call__(self, x: NDArray[Any], _var: float, /) -> NDArray[Any]:
+        """Return *x* unchanged."""
+        return x
+
+
+@dataclass
+class Lognormal:
+    r"""
+    Transformation for lognormal fields.
+
+    .. math::
+
+       t(X) = \lambda \bigl[\exp(X - \tfrac{\sigma^2}{2}) - 1\bigr]
+
+    The "lognormal shift" parameter :math:`\lambda` is the scale
+    parameter of the distribution, with determines the smallest
+    possible value :math:`-\lambda` of the lognormal field.
+
+    """
+
+    lamda: float = 1.0
+
+    def __call__(self, x: NDArray[Any], var: float, /) -> NDArray[Any]:
+        """Transform *x* into a lognormal field."""
+        xp = x.__array_namespace__()
```

Comment:
I see a bit of Array API stuff. I know we (@Saransh-cpp and I) aren't meant to be touching that at the moment. Guess it's fine for you to?

Reply:
Yes, this is all fully Array API-compliant (Saransh asked me to use NumPy types here). The reason this needs to be merged now is for a project not using NumPy.

Reply:
The entire code is actually Array API-compliant (Saransh wanted to use NDArray here). This needs to be merged for a project that isn't using NumPy, in fact